libs/common-system/src/main/java/org/codice/ddf/configuration/SystemBaseUrl.java

```diff
@@ -39,17 +53,51 @@
 
   public static final String DEFAULT_PROTOCOL = "https://";
 
-  private SystemBaseUrl() {}
+  /* Accessor Fields to be set based on External or Internal*/
+  private String httpPort;
+
+  private String httpsPort;
+
+  private String port;
+
+  private String host;
+
+  private String protocol;
+
+  /* EXTERNAL and INTERNAL singleton objects */
+  public static final SystemBaseUrl EXTERNAL = new SystemBaseUrl(true);
+  public static final SystemBaseUrl INTERNAL = new SystemBaseUrl(false);
+
+  private SystemBaseUrl(boolean external) {
+    if (external) externalMode();
+    else internalMode();
+  }
+
+  private void externalMode() {
+    httpPort = EXTERNAL_HTTP_PORT;
+    httpsPort = EXTERNAL_HTTPS_PORT;
+    port = EXTERNAL_PORT;
+    host = EXTERNAL_HOST;
+    protocol = EXTERNAL_PROTOCOL;
+  }
+
+  private void internalMode() {
+    httpPort = HTTP_PORT;
+    httpsPort = HTTPS_PORT;
+    port = PORT;
+    host = HOST;
+    protocol = PROTOCOL;
+  }
 
   /**
    * Gets the port number based on the the system protocol
    *
    * @return
    */
-  public static String getPort() {
-    String port = System.getProperty(PORT);
+  public String getPortString() {
```

Comment:
❗️ **Currently,** I'm using two different methods for internal and external properties. Should I refactor this so they only use one common method?

Reply:
The two different methods are the singleton methods and legacy static accessors. Eliminating the redundancy would be large refactoring, and perhaps should be a separate piece of tech debt.